src/components/DownloadLimitEntry.vue

```diff
@@ -0,0 +1,186 @@
+				class="action__count-note"
+				type="info">
+				{{ n('files_downloadlimit', '1 remaining download allowed', '{count} remaining downloads allowed', remainingCount, { count: remainingCount }) }}
+			</NcNoteCard>
+			<NcTextField v-show="limitEnabled"
+				:label="t('files_downloadlimit', 'Set download limit')"
+				type="number"
+				min="1"
+				:value.sync="limit"
+				:helper-text="helperText"
+				:error="Boolean(helperText)" />
+			<NcNoteCard v-show="limitEnabled && showResetNote"
+				class="action__reset-note"
+				type="warning">
+				{{ t('files_downloadlimit', 'Setting a new limit will reset the download count') }}
+			</NcNoteCard>
+		</template>
+	</Fragment>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import { Fragment } from 'vue-frag'
+import { showError } from '@nextcloud/dialogs'
+
+import NcCheckboxRadioSwitch from '@nextcloud/vue/dist/Components/NcCheckboxRadioSwitch.js'
+import NcNoteCard from '@nextcloud/vue/dist/Components/NcNoteCard.js'
+import NcTextField from '@nextcloud/vue/dist/Components/NcTextField.js'
+
+import {
+	getDownloadLimit,
+	setDownloadLimit,
+	deleteDownloadLimit,
+} from '../services/DownloadLimitService.ts'
+import { logger } from '../logger.ts'
+
+export default defineComponent({
+	name: 'DownloadLimitEntry',
+
+	components: {
+		Fragment,
+		NcCheckboxRadioSwitch,
+		NcNoteCard,
+		NcTextField,
+	},
+
+	props: {
+		share: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			limitEnabled: false,
+			initialLimit: null,
+			limit: '', // input should be empty at first so initialize as empty string
+			count: null,
+			loading: false,
+			hasError: false,
+		}
+	},
+
+	computed: {
+		remainingCount() {
+			return this.initialLimit - this.count
+		},
+
+		helperText() {
+			if (this.limit > 0) {
+				return ''
+			}
+			return t('files_downloadlimit', 'The minimum limit is 1')
+		},
+
+		isNewShare() {
+			return this.share.id === null || this.share.id === undefined
+		},
+
+		showRemainingDownloadsNote() {
+			return typeof this.count === 'number'
+		},
+
+		showResetNote() {
+			return !this.isNewShare
+				&& Boolean(this.initialLimit)
+				&& typeof this.limit === 'number'
+				&& this.limit !== this.initialLimit
+		},
+	},
+
+	watch: {
+		limit(limit) {
+			this.limit = Number(limit) // emitted <input> value is string so we parse it to number
+		},
+	},
+
+	async created() {
+		logger.debug('Loading download limit', { share: this.share })
+		this.loading = true
+		try {
+			const { limit, count } = await getDownloadLimit(this.share.token)
+			if (typeof limit === 'number' && typeof count === 'number') {
+				this.limitEnabled = Boolean(limit)
+				this.initialLimit = limit
+				this.limit = limit
+				this.count = count
+			}
+		} catch (error) {
+			this.hasError = true
+			logger.error('Failed to load download limit', { error, share: this.share })
+		}
+		this.loading = false
+	},
+
+	methods: {
+		async onSave() {
```

Comment:
Doesn't seem to be used anywhere?

Reply:
Was discussed doing it this "simpler" way 🙈, it is called in server